src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -3,16 +3,1430 @@
+        }
+
+        if (WI_IsFlagSet(modifiers, Settings::KeyModifiers::Shift))
+        {
+            buffer += L"Shift+";
+        }
+
+        if (WI_IsFlagSet(modifiers, Settings::KeyModifiers::Alt))
+        {
+            buffer += L"Alt+";
+        }
+
+        return buffer;
+    }
+
+    // Method Description:
+    // - Takes a MenuFlyoutItem and a corresponding KeyChord value and creates the accelerator for UI display.
+    //   Takes into account a special case for an error condition for a comma
+    // Arguments:
+    // - MenuFlyoutItem that will be displayed, and a KeyChord to map an accelerator
+    void TerminalPage::_SetAcceleratorForMenuItem(Controls::MenuFlyoutItem& menuItem,
+                                                  const winrt::Microsoft::Terminal::Settings::KeyChord& keyChord)
+    {
+#ifdef DEP_MICROSOFT_UI_XAML_708_FIXED
+        // work around https://github.com/microsoft/microsoft-ui-xaml/issues/708 in case of VK_OEM_COMMA
+        if (keyChord.Vkey() != VK_OEM_COMMA)
+        {
+            // use the XAML shortcut to give us the automatic capabilities
+            auto menuShortcut = Windows::UI::Xaml::Input::KeyboardAccelerator{};
+
+            // TODO: Modify this when https://github.com/microsoft/terminal/issues/877 is resolved
+            menuShortcut.Key(static_cast<Windows::System::VirtualKey>(keyChord.Vkey()));
+
+            // inspect the modifiers from the KeyChord and set the flags int he XAML value
+            auto modifiers = AppKeyBindings::ConvertVKModifiers(keyChord.Modifiers());
+
+            // add the modifiers to the shortcut
+            menuShortcut.Modifiers(modifiers);
+
+            // add to the menu
+            menuItem.KeyboardAccelerators().Append(menuShortcut);
+        }
+        else // we've got a comma, so need to just use the alternate method
+#endif
+        {
+            // extract the modifier and key to a nice format
+            auto overrideString = _FormatOverrideShortcutText(keyChord.Modifiers());
+            auto mappedCh = MapVirtualKeyW(keyChord.Vkey(), MAPVK_VK_TO_CHAR);
+            if (mappedCh != 0)
+            {
+                menuItem.KeyboardAcceleratorTextOverride(overrideString + gsl::narrow_cast<wchar_t>(mappedCh));
+            }
+        }
+    }
+
+    // Method Description:
+    // - Place `copiedData` into the clipboard as text. Triggered when a
+    //   terminal control raises it's CopyToClipboard event.
+    // Arguments:
+    // - copiedData: the new string content to place on the clipboard.
+    void TerminalPage::_CopyToClipboardHandler(const winrt::hstring& copiedData)
+    {
+        this->Dispatcher().RunAsync(CoreDispatcherPriority::High, [copiedData]() {
+            DataPackage dataPack = DataPackage();
+            dataPack.RequestedOperation(DataPackageOperation::Copy);
+            dataPack.SetText(copiedData);
+            Clipboard::SetContent(dataPack);
+
+            // TODO: MSFT 20642290 and 20642291
```

Comment:
I believe this is also missing code from #2486 and #1224

Reply:
I'm realizing now that you have yet to merge master for those updates - they'll be included in that merge